Python/bytecodes.c

```diff
@@ -336,20 +336,37 @@ dummy_func(
             res = PyStackRef_NULL;
         }
 
-        macro(END_FOR) = POP_TOP;
+        inst(END_FOR, (value -- )) {
+            /* Don't update instr_ptr, so that POP_ITER sees
+             * the FOR_ITER as the previous instruction.
+             * This has the benign side effect that if value is
+             * finalized it will see the location as the FOR_ITER's.
+             */
+            frame->instr_ptr = prev_instr;
```

Comment:
Is there a precedent for this? Seems like the kind of thing that can cause confusion/problems later on.

Reply:
The tier 2 code generator doesn't like it either. I think I'll need to add a new annotation "no_saved_ip" to tell the code generator not to update `frame->instr_ptr`.

Any suggestions for a better name than "no_saved_ip"?